src/Core/ScreenshotCache.vala

```diff
@@ -0,0 +1,201 @@
+public class AppCenterCore.ScreenshotCache {
+    private const int MAX_CACHE_SIZE = 100000000;
+
+    public string screenshot_path;
+
+    // Atomic integer for keeping track of the current screenshot usage.
+    private int screenshot_usage;
+
+    public ScreenshotCache () {
+        screenshot_path = GLib.Environment.get_user_cache_dir ()
+            + "/"
+            + Build.PROJECT_NAME
+            + "/screenshots";
+
+        if (GLib.DirUtils.create_with_parents (screenshot_path, 0755) == -1) {
+            critical (
+                "Error creating the temporary folder: GFileError #%d",
+                GLib.FileUtils.error_from_errno (GLib.errno)
+            );
```

Comment:
Should probably return at that point?

Reply:
And I don't think we should crash here, just not use cache instead

Reply:
@Philip-Scott does critical crash by default? I didn't think it did.

Reply:
Critical and error crash, doesn't it? If not we can always change it to warning

Reply:
I've changed it to warning, and will now return a null pointer in the event that it couldn't be retrieved.